frontend/views/containers/sidebar/GroupsMinIncome.vue

```diff
@@ -10,27 +10,52 @@ div
     :aria-label='L("Change Mincome")'
     @click='openProposal'
   ) Change
+  br
+  i18n.link(
+    tag='button'
+    @click='changeMin'
+  ) Increase {{ currency }}10
 </template>
 
 <script>
 import sbp from '~/shared/sbp.js'
 import { OPEN_MODAL } from '@utils/events.js'
 import currencies from '@view-utils/currencies.js'
-import { mapGetters } from 'vuex'
+import { mapState, mapGetters } from 'vuex'
 
 export default {
   name: 'GroupsMinIncome',
   computed: {
     currency: function () {
       return currencies[this.groupSettings.incomeCurrency]
     },
+    ...mapState([
+      'currentGroupId'
+    ]),
     ...mapGetters([
       'groupSettings'
     ])
   },
   methods: {
     openProposal () {
       sbp('okTurtles.events/emit', OPEN_MODAL, 'MincomeProposal')
+    },
+    async changeMin () {
+      const updatedSettings = await sbp('gi.contracts/group/updateSettings/create',
+        {
+          incomeProvided: this.groupSettings.incomeProvided + 10
+        },
+        this.currentGroupId // NOTE 1: why do i need to pass currentGroupId?
+        // Actually, I understand why, to give context... but, how would I know it?
+        // When reading the selector at group.js, that's expecting state, data, meta,
+        // but not a groupId... It took me a **while** to realize that
+        // when I call this selector, it actually calls Contract.js 'selectors/register'
+        // first and only then it calls GIMessage.create() (what for?)
+      )
+
+      // NOTE: I was expecting "updateSettings/process" to be called by now...
+      // but it didn't happen. Then I realized I also had to call publishLogEntry. Why?
```

Comment:
It's unclear to me why you're asking this question, as the answer is evident from reading the implementation of `'gi.contracts/group/updateSettings/create'` and `'backend/publishLogEntry'`. I guess what I'm saying is that I'm surprised by the question itself, and that if it's still not clear we should do a call to see how you're reading the codebase, as the problem is likely there

Reply:
Greg, once again, I understand that this is all obvious for you (you were the only creator of it!), but it isn't for anyone else in the project at the moment.

So, when someone asks you something, for as dumb as it might sound to you, your answer saying _"it's evident"_ or _"just read the file"_ isn't helpful at all.

Of course, I already read all of those files. But there's a difference between reading code and understanding exactly how it works and why. That's the part I expect from you a little more of patience and clarification in your written words. Don't be _surprised_ by any type of question. This is the normal process of learning something, I guess... Tentative, failure and asking dumb questions. Especially when we have zero proper documentation about how this whole thing works.

So, I'll try to ask this again:

Can you explain the flow behind these two lines and why it works like that? Why do we need 2 commands? In what scenarios it's helpful to just do `create` or just call `publishLogEntry`?

```js
const updatedSettings = await sbp('gi.contracts/group/updateSettings/create')
await sbp('backend/publishLogEntry', updatedSettings)
```

Reply:
@sandrina-p Fair points all around! I think after our call this is now all clearer to you, so perhaps while you still have your "beginner vision" you might be able to help others by updating our documentation a little. At least you are welcome to do so if you so feel inclined!

Reply:
Will do it 👍